### ISSUE TITLE
[Feature] `.$` and `.$$` for locators? 

### ISSUE THREAD

**Issue by mgenware** [microsoft/playwright#15044]
I'm migrating a playwright codebase using `ElementHandle` to locators. I really miss the `$` and `$$`. I found locator code tends to be longer and also requires you to pay a bit more attention not to fall into runtime errors.

Some examples:

```js
// Example 1
// === ElementHandle
element.$(sel).click();
// === Locator
// Might cause runtime error when there're more than 1 matches in strict mode.
element.locator(sel).click();
// Better to select the first child.
element.locator(sel).nth(0).click();

// Example 2
// Let's say there's a bug in your code where you accidentally called `click` on a list of elements.
// === ElementHandle
(await element.$$(listSel)).click(); // Compile error! You can't call click() on an array.
// === Locator
element.locator(listSel).click(); // Compiles fine. If it happens to get a list with only 1 child, it also runs fine.
```

Would it be possible to add 2 helper functions like `.locate$` and `.locate$$` for locators? thanks!

**Comment by mxschmitt** [comment]
Could you provide us examples how these functions should behave? Its not very clear to me.

Seems a bit related to https://github.com/microsoft/playwright/issues/12336?

**Comment by mgenware** [comment]
That makes sense. Tho it might introduce some `nth(0)` boilerplate in some circumstances. Like the one I ran into, where we have nested comments (like reddit), the auto shadow dom piercing feature makes `locator` calls always end up with a bunch of elements from all its descendants.. we're sure the first matched one is what we need. Therefore, a lot of `nth(0)` calls were added after `locator` (previously using `$` without issues). Overall, not a big deal.
